src/DataFramesMeta.jl

```diff
@@ -392,10 +391,35 @@ function transform(g::GroupedDataFrame; kwargs...)
     idx1 = [1; 1 .+ idx2[1:end-1]]
     for (k, v) in kwargs
         first = v(g[1])
-        result[k] = Array{eltype(first)}(undef, size(result, 1))
-        result[idx1[1]:idx2[1], k] = first
-        for i in 2:length(g)
-            result[idx1[i]:idx2[i], k] = v(g[i])
+        if first isa AbstractVector 
+            if length(first) != size(g[1], 1)
+                throw("If a function returns a vector, the result " * 
+                      "must have the same length as the groups it " *
+                      "operates on")
+            end
+            result[k] = Array{eltype(first)}(undef, size(result, 1))
```

Comment:
It should be faster to also store `result[k]` in a variable here and work directly on it below, as a data frame lookup is a bit costly. You'd also need to do that when reallocating the vector of course.

Reply:
fixed